### PR TITLE
County Not Listed

### PR DESCRIPTION
**Pull request by bengolder**
### Expected issues this will close:

- [x] #1099 Create failing browser tests for CNL workflows
    - [x] Apply to CNL only
    - [x] Apply to CNL + partner
    - [x] CMR staff alert and respond
- [x] #1100 Edit County and Organization Fixtures
- [x] #1102 Create FormSpec & Fields for CNL
- [x] #1101 Create dynamic confirmation messages
- [x] #929 Applicant can mark that their county is not listed
- [x] #938 Remove county affirmation checkbox
- [x] #932 Applicant can can essential info for county not listed
- [x] #930 Applicant can input the counties they need help in
- [x] #933 Applicant can apply to partner county + not listed
- [x] #934 CNL-only confirmation page
- [x] #935 CNL + partners confirmation page
- [x] #936 Alerts for CMR staff about CNL submissions
- [x] #931 CMR staff can view all CNL submissions
- [x] #937 Track CNL submission in Mixpanel (this will happen by default, with no changes, because submissions to any county are tracked)

### What does this do and why?

This will allow applicants to apply for counties where we don't yet have partners and will allow CMR staff to review these applications in response to alerts.

### Testing & Checks

What does this PR include?
- [ ] new dependencies
- [ ] migrations
- [x] data migrations
- [ ] celery tasks
- [ ] changes to environment variables or local settings
- [ ] configuration changes for 3rd party services (Travis, Heroku, AWS, Github, etc) 
- [x] visual or style changes

#### Important UAT Checks

- [x] (Role: Applicant) Apply for one or more unlisted counties
    - [x] Review county selection page
    - [x] Review main application (ensure fields are okay)
    - [x] Review confirmation page
    - [x] Review email & text message confirmations
- [x] (Role: Applicant) Apply for one or more unlisted counties as well as a partner county.
    - [x] Review county selection page
    - [x] Review main application (ensure fields are okay)
    - [x] Review confirmation page
    - [x] Review email & text message confirmations
- [x] (Role: CMR Staff) Review & handle CNL apps
    - [x] Ensure that you can see CNL apps
    - [x] add the tag `cnl_done` to an app

#### New automated tests
- This will add new browser tests for applying to CNL-only and CNL+partner, as well as for CMR staff to review.

#### Managing CNL apps & whether they have been read

- The County-Not-Listed Index page will show _all_ county not listed apps in descending order of date submitted.
- Every weekday, we will send out a count of all _unhandled_ county not listed apps in a slack notification.
- **_unhandled_ CNL apps will be defined by _not_ having the `cnl_done` tag**. So we can mark that these are done by adding the `cnl_done` tag. It is easy to change the name of this tag, and we should confirm this tag is what we want before we deploy
- if an app has the `cnl_done` tag, it will no longer be included in the count for the slack notification, but it _will_ still appear on the County-Not-Listed index page

#### Mixpanel & Metrics changes

- We will see CNL apps immediately in some existing mixpanel charts. For example, in "Application Submitted by organizations" on the dashboard, they will be visible as applications destined for "Code for America"
- CNL applications will be evident in mixpanel through two properties `counties=not_listed` (on the `application_started` event) and `application_organization=Code for America` for any events from `application_submitted` or later. In other words, we can create a funnel specific to CNL applications, and in most org-based breakdowns of applications, they will be visible as applications to `Code for America`

#### Refactoring of constants and county choices

As part of this PR, we removed many redundant hardcoded variables from `intake/constants.py`, substituting queries and string literals where appropriate. There were a few critical references to these hardcoded variables in application code and many non-critical references throughout unit tests.

A portion of this refactor touched old migrations. There were data migrations that imported and used these constants. Those references were replaced with string literals matching the imported constants.

#### Visual and copy changes

_New county selection page_
screen shot 2017-08-31 at 4 15 03 pm

_CNL-specific application page_
screen shot 2017-08-31 at 4 15 16 pm

_CNL-specific confirmation page_
screen shot 2017-08-31 at 4 17 20 pm

_New tab in application index_
screen shot 2017-09-07 at 11 47 29 am

_County Not Listed index page_
screen shot 2017-09-07 at 11 47 53 am

### CODE REVIEW

**Comment by TiffanyAndrews** [comment]
UAT Feedback
@bengolder 
_General _
- When the `text` contact checkbox  is marked and a phone number and email is provided, the confirmation message is sent to the email and phone. Is email the default or is this an error?
- Change content: `In counties where we don't have official partners, we will send you information on how to get started. 


_CNL ONLY Option_
-  This question is confusing for CNL only option.  Is connected to automated messaging? Can it be removed and the applicant still receives a confirmation message via email and text?


_CNL & Listed County Option_
- When `Alameda, or Contra Costa and CNL ` selected, the is the top message list `Alameda` or `Contra Costa` first then CNL message


 However, all the other counties ( `Fresno -- Yolo`) are listed after the  `CNL ` message

 
If `Alameda` or `Contra Costa`, `CNL` and `another county`,  CNL is sandwiched in between the selected counties.

File: intake/tests/models/test_form_submission.py [hunk 1]
```diff
@@ -82,33 +80,6 @@ def test_get_transfer_action_returns_none(self):
         self.assertIsNone(
             submission.get_transfer_action(request))
 
-    def test_qualifies_for_fee_waiver_with_public_benefits(self):
```
Comment (rossettistone) [inline intake/tests/models/test_form_submission.py:85]: @bengolder Can you recall why we removed these tests? Do they refer to a defunct feature?

Comment (bengolder) [reply intake/tests/models/test_form_submission.py:85]: I moved this functionality and this test into the submissions service.

**Comment by TiffanyAndrews** [comment]
@bengolder @rossettistone no changes. ✔️ UAT. 👍 🚀

**Comment by bengolder** [comment]
We seem to be having persistent troubles with Travis, and the problems seem to be specific to testing environments. After discussing with @rossettistone and @glassresistor, I'm going to merge this despite failing Travis checks.

**Pull request merged**